client/web/src/repo/blob/BlobLoadingSpinner.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+
+import { useLocation } from 'react-router'
+
+import { Link, LoadingSpinner } from '@sourcegraph/wildcard'
+
+export function BlobLoadingSpinner(): JSX.Element {
+    const [afterOneSec, setAfterOneSec] = useState(false)
+    const [afterThreeSec, setAfterThreeSec] = useState(false)
+    const [afterSixSec, setAfterSixSec] = useState(false)
+    const [afterNineSec, setAfterNineSec] = useState(false)
+
+    useEffect(() => {
+        const timeout = setTimeout(() => setAfterOneSec(true), 1000)
+        return () => clearTimeout(timeout)
+    }, [])
+    useEffect(() => {
+        const timeout = setTimeout(() => setAfterThreeSec(true), 3000)
+        return () => clearTimeout(timeout)
+    }, [])
+    useEffect(() => {
+        const timeout = setTimeout(() => setAfterSixSec(true), 6000)
+        return () => clearTimeout(timeout)
+    }, [])
+    useEffect(() => {
+        const timeout = setTimeout(() => setAfterNineSec(true), 9000)
+        return () => clearTimeout(timeout)
+    }, [])
+
+    const location = useLocation()
+
+    return (
+        <div className="d-flex mt-3 justify-content-center">
+            {afterOneSec ? (
+                <div className="d-flex flex-column align-items-center">
+                    <LoadingSpinner />
+                    <div className="text-muted mt-2">
+                        {afterNineSec ? (
+                            <>
+                                It’s taking much longer than expected to load this file. Try{' '}
+                                <Link to={location.pathname} onClick={reload}>
+                                    reloading the page
+                                </Link>
```

Comment:
```suggestion
                                <Link to={location.pathname} onClick={reload}>
                                    reloading the page
                                </Link>.
```

Adding a period here so both statements are consistent looks more natural to me.